nntrainer/graph/graph_node.h

```diff
@@ -84,7 +84,7 @@ class GraphNode {
 template <typename T>
 class GraphNodeIterator : public std::iterator<std::random_access_iterator_tag,
                                                std::shared_ptr<GraphNode>> {
-  std::shared_ptr<GraphNode> *p; /** underlying object of GraphNode */
```

Comment:
How about NodeIterator wraps around the vector iterator and all the typedefs are delegated from it?

Reply:
The typedefs cant be removed as `const shared_ptr<GraphNode>` and `shared_ptr<const GraphNode>` are different, and same for LayerNode as well.

Reply:
Okay